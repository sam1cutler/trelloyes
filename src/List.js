import React from 'react';



function List(props) {
    console.log('Lis.js ran, with the following props:');
    console.log(props);
}


// const arrayOfLists = 