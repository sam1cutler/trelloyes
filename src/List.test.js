import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

const sampleCards = [
    {id: 'b', title: 'Second card', content: 'lorem ipsum' },
    { id: 'c', title: 'Third card', content: 'lorem ipsum' },
    { id: 'd', title: 'Fourth card', content: 'lorem ipsum' },
    { id: 'f', title: 'Sixth card', content: 'lorem ipsum' },
    { id: 'h', title: 'Eighth card', content: 'lorem ipsum' },
    { id: 'i', title: 'Ninth card', content: 'lorem ipsum' },
    { id: 'k', title: 'Eleventh card', content: 'lorem ipsum' }
]

describe('List component', () => {
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List header='Second list' cards={sampleCards}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the List component as expected with the "Second List" cardlist.', () => {
        const tree = renderer
          .create(<List header='Second list' cards={sampleCards}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

})