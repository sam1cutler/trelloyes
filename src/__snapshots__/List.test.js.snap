// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the List component as expected with the "Second List" cardlist. 1`] = `
<div>
  <header
    className="List-header"
  >
    <h2>
      Second list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <ul>
      <li>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </li>
      <li>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Third card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </li>
      <li>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fourth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </li>
      <li>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </li>
      <li>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eighth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </li>
      <li>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Ninth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </li>
      <li>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eleventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </li>
    </ul>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</div>
`;
